pkg/security/resolvers/dns/resolver.go

```diff
@@ -0,0 +1,158 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+// Package dns resolves ip addresses to hostnames
+package dns
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/security/metrics"
+	"github.com/DataDog/datadog-agent/pkg/security/probe/config"
+	"github.com/DataDog/datadog-go/v5/statsd"
+	lru "github.com/hashicorp/golang-lru/v2"
+	"go.uber.org/atomic"
+	"net/netip"
+)
+
+// CacheStats defines metrics for the LRU
+type CacheStats struct {
+	cacheHits       atomic.Int64
+	cacheMisses     atomic.Int64
+	cacheInsertions atomic.Int64
+	cacheEvictions  atomic.Int64
+}
+
+// Resolver defines a DNS resolver
+type Resolver struct {
+	cache         *lru.Cache[netip.Addr, map[string]bool]
+	cnameCache    *lru.Cache[string, map[string]bool]
```

Comment:
I'm not sure why we have to use a map[string]bool instead of a slice of []string here ? It would simplify the code below